azure-proto-compute/VirtualMachine.cs

```diff
@@ -18,9 +24,16 @@ internal VirtualMachine(ResourceOperationsBase operations, VirtualMachineData re
             Data = resource;
         }
 
-        /// <summary>
-        /// Gets the data representing this VirtualMachine.
-        /// </summary>
-        public VirtualMachineData Data { get; private set; }
+        /// <inheritdoc />
+        protected override VirtualMachine GetResource()
```

Comment:
I think we need to add this overload to all [Resource].cs classes otherwise StartAddTag for all other types will make an extra network call that isn't needed